src/windows/enumerate.rs

```diff
@@ -504,7 +549,7 @@ fn test_parsing_usb_port_information() {
 
     assert_eq!(info.vid, 0x0403);
     assert_eq!(info.pid, 0x6001);
-    assert_eq!(info.serial_number, Some("A702TB52A".to_string()));
+    assert_eq!(info.serial_number, Some(r"A702TB52A\0000".to_string()));
```

Comment:
With the new impl, if I interpreted what you & the pyserial regex mean correctly, it just matches to the end of the string. Meaning that this is the serial number it interprets now.

Reply:
Wow, you are quick! Yes, pySerial's regex matches until the end of the string. But they are distinguishing between the two bus types `USB` and `FTDIBUS` for the former we are sorted. But for the latter we need to stop matching at the end of the middle paths segment and ignore `\0000`.

Reply:
May be splitting the string at first at `\`, then deciding how to interpret middle path element according to the first one (`FTDIBUS` and everything else) and then handle things according to whether we are dealing with the interface of a composite device or the device itself.